back-end/www/models/model_operations/answer_operations.py

```diff
@@ -0,0 +1,306 @@
+            box_bottom_right_lat=bbox_bottom_right_lat,
+            bbox_bottom_right_lng=bbox_bottom_right_lng, zoom_level=zoom_level)
+
+    db.session.add(answer)
+    db.session.commit()
+
+    return answer
+
+
+def create_answer_with_result(user_id, location_id, year_old, year_new,
+        source_url_root, land_usage, expansion, is_gold_standard,
+        bbox_left_top_lat, bbox_left_top_lng, bbox_bottom_right_lat,
+        bbox_bottom_right_lng, zoom_level):
+    """
+    Create an answer and return if it passed or not by checking the gold standards.
+
+    Parameters
+    ----------
+    user_id : int
+        Foreign key to the user table.
+    location_id : int
+        Foreign key to the location table.
+    year_old: int
+        The year where the satellite photo was taken from our geo sources.
+    year_new: int
+        A newer photo to be compared with the one taken in year_old.
+    source_url_root : str
+        URL to store the location on the map.
+    land_usage : int
+        User's answer of judging if the land is a farm or has buildings.
+        (check the answer table in model.py for the meaning of the values)
+    expansion : int
+        User's answer of judging the construction is expanded.
+        (check the answer table in model.py for the meaning of the values)
+    is_gold_standard : bool
+        If it's a golden standard answer provided by admnin.
+    bbox_left_top_lat : float
+        The latitude of the top-left corner of the bounding box for displaying the focus.
+    bbox_left_top_lng : float
+        The longitude of the top-left corner of the bounding box for displaying the focus.
+    bbox_bottom_right_lat : float
+        The latitude of the bottom-right corner of the bounding box for displaying the focus.
+    bbox_bottom_right_lng : float
+        The longitude of the bottom-right corner of the bounding box for displaying the focus.
+    zoom_level : int
+        The zoom level for displaying the location.
+
+    Returns
+    -------
+    compare_result : int
+        Compare with the gold answer.
+        0 means Pass
+        1 means Fail
+        -1 means that the gold standard does not exist
+
+    Raises
+    ------
+    exception : Exception
+        Type checking for land_usage, expansion, year_old, year_new, and is_gold_standard.
+    """
+    new_answer = create_answer(user_id, location_id, year_old, year_new,
+            source_url_root, land_usage, expansion, is_gold_standard,
+            bbox_left_top_lat, bbox_left_top_lng, bbox_bottom_right_lat,
+            bbox_bottom_right_lng, zoom_level)
+
+    # Bypass the gold standard itself
+    if(is_gold_standard):
+        return 0
+
+    # Get the gold answer and compare result
+    compare_result = 0
+    gold_answer = Answer.query.filter_by(is_gold_standard=True,
+            location_id=location_id, year_old=year_old, year_new=year_new).first()
+    if(gold_answer != None):
```

Comment:
Keep the syntax consistent. In python, there is no need to put the () after the if statement. You can just do
```
 if gold_answer != None:
```

Reply:
Thanks for the kind remind for the python newbie :-)